hbase-client/src/main/java/org/apache/hadoop/hbase/client/replication/ReplicationPeerConfigUtil.java

```diff
@@ -455,36 +458,45 @@ public static ReplicationPeerConfig appendTableCFsToReplicationPeerConfig(
 
   /**
    * Helper method to add base peer configs from Configuration to ReplicationPeerConfig
-   * if not present in latter.
    *
    * This merges the user supplied peer configuration
    * {@link org.apache.hadoop.hbase.replication.ReplicationPeerConfig} with peer configs
    * provided as property hbase.replication.peer.base.configs in hbase configuration.
-   * Expected format for this hbase configuration is "k1=v1;k2=v2,v2_1". Original value
-   * of conf is retained if already present in ReplicationPeerConfig.
+   * Expected format for this hbase configuration is "k1=v1;k2=v2,v2_1".
    *
    * @param conf Configuration
    * @return ReplicationPeerConfig containing updated configs.
    */
-  public static ReplicationPeerConfig addBasePeerConfigsIfNotPresent(Configuration conf,
+  public static ReplicationPeerConfig updateReplicationBasePeerConfigs(Configuration conf,
     ReplicationPeerConfig receivedPeerConfig) {
-    String basePeerConfigs = conf.get(HBASE_REPLICATION_PEER_BASE_CONFIG, "");
+    String removeBasePeerConfigs = conf.get(HBASE_REPLICATION_PEER_REMOVE_BASE_CONFIG, "");
     ReplicationPeerConfigBuilder copiedPeerConfigBuilder = ReplicationPeerConfig.
       newBuilder(receivedPeerConfig);
-    Map<String,String> receivedPeerConfigMap = receivedPeerConfig.getConfiguration();
 
+    // remove the peer configurations specified in the conf
```

Comment:
Because configuration settings are often from hbase-site.xml, and hbase-site.xml can be different on different region servers (e.g when upgrading a cluster), it would seem like you could get into some unwelcome race conditions with this approach. 

Let's say that you're doing a rolling restart from Version 1 to Version 2. Version 1 has a config C == true, Version 2 tries to remove C. 

1. Server 1 is upgraded to Version 2 and restarts, and removes C from the peer config
2. Server 2 is still on Version 1 and is restarted for an unrelated reason (maybe it crashes and is restarted). It sees that C is not in the peer config, and recreates it. 

Assuming that you upgraded hbase-site.xml everywhere and restarted, the cluster would _eventually_ converge on the right answer, but there's a lot of opportunity for the config to flicker back and forth in the meantime. Depending on the config, that could have incorrect or unpredictable results

Reply:
That's a good point. Ideally IMO, the peer configuration should also not be added from the region server restart. We should have a better home for this (may be master restart?). 

@virajjasani  Do you think if it is standard to have conf upgrade before the HBase rolling upgrade? If not, we might need to rework on updating the peer configs

Reply:
Also, looking at the `master` branch code, looks like the RS restart has nothing to do with this update config. Only master restart is able to update configs.

Reply:
Sorry just got to this PR. Thats right, master branch replication coordination is different from branch-1. This was specifically taken into consideration in the original PR that added this feature (see the commit message for branch-1)..